### ISSUE TITLE
New file with same name as recently renamed file fails to save

### ISSUE THREAD

**Issue by aaronadamsTO** [microsoft/vscode-remote-release#2210]
- VSCode Version: 1.41.1
- Local OS Version: Windows_NT x64 10.0.18363
- Remote OS Version: Debian 10.2
- Remote Extension/Connection Type: Docker

### Steps to reproduce

1. Rename config.yml to config.old.yml
2. Create new file config.yml
3. Edit config.yml and save
4. Edit config.yml and save again
5. Edit config.yml and save a third time

This occurs only when editing via Remote-Containers, I cannot reproduce it locally.

### What should happen

config.yml saves.

### What actually happens

> Failed to save 'config.yml': The content of the file is newer. Please compare your version with the file contents.

### Notes

(Note I previously filed #2160 and then closed it as a Docker Desktop issue - this seems to be different behaviour I'm now observing, since the Docker container's time is accurate. #2165 is another similar issue.)

A container shell shows that my container time is accurate, and confirms I am attempting to overwrite an older file:

```sh
$ ls -l config.yml
-rwxr-xr-x 1 root root 1047 Jan 27 16:56 config.yml
$ date
Mon Jan 27 17:13:08 UTC 2020
```

Window log entry:

```
[2020-01-27 12:14:30.570] [renderer1] [error] doSave(7) - exit - resulted in a save error: Error: Unable to write file (Error: File Modified Since) {"$mid":1,"fsPath":"\\workspaces\\XXX\\.circleci\\config.yml","_sep":1,"external":"vscode-remote://dev-container%2B633a5c55736572735c4161726f6e4164616d735c5265706f7369746f726965735c63697263756c6172732d72657461696c65722d646f6373/workspaces/XXX/.circleci/config.yml","path":"/workspaces/XXX/.circleci/config.yml","scheme":"vscode-remote","authority":"dev-container+633a5c55736572735c4161726f6e4164616d735c5265706f7369746f726965735c63697263756c6172732d72657461696c65722d646f6373"}
```

My Dockerfile is very straightforward:

```
FROM mcr.microsoft.com/vscode/devcontainers/base:debian-10

# Avoid warnings by switching to noninteractive
ENV DEBIAN_FRONTEND=noninteractive

# Configure apt and install packages
RUN apt-get update \
  #
  # Install Ruby
  && apt-get -y install --no-install-recommends make build-essential ruby ruby-dev \
  #
  # Install Bundler
  && gem install bundler \
  #
  # Install CircleCI CLI
  && curl -fLSs https://circle.ci/cli | bash \
  #
  # Clean up
  && apt-get autoremove -y \
  && apt-get clean -y \
  && rm -rf /var/lib/apt/lists/*

# Switch back to dialog for any ad-hoc use of apt-get
ENV DEBIAN_FRONTEND=dialog
```

**Comment by aaronadamsTO** [comment]
I found a reproducible case; it involves moving a file out of the way, creating a new file in its place, and editing it. Editing always seems to fail on the third save.

This occurs only when editing via Remote-Containers, I cannot reproduce it locally.

### Steps to reproduce

1. Rename config.yml to config.old.yml
2. Create new file config.yml
3. Edit config.yml and save
4. Edit config.yml and save again
5. Edit config.yml and save a third time

### What should happen

config.yml saves.

### What actually happens

> Failed to save 'config.yml': The content of the file is newer. Please compare your version with the file contents.

**Comment by aaronadamsTO** [comment]
I updated the original ticket with the new findings.

**Comment by roblourens** [comment]
I can't reproduce this. Can you check the actual contents of the file, and are they actually different? 
Is there some extension or script changing this file in the background or does this happen with any arbitrary file?

**Comment by aaronadamsTO** [comment]
Yeah, it's a bit confusing. It pops up pretty consistently for me after saving a file three times; I can work around it now by closing and re-opening the file tab, at which point I generally get another few saves.

Per my original report, I initially suspected a Docker Desktop bug, but then thought I'd ruled it out; now I'm wondering again if it's the root cause. Here's the issue, now in their latest stable release:

https://github.com/docker/for-win/issues/5593

Basically the container time pauses during sleep, and doesn't refresh on resume.

I wonder if internally you're using a mix of local time and container time in the comparison? If so, I suppose anytime the two aren't synchronized you'd have problems.

Docker's getting a ton of complaints about this release, so I'm assuming they'll release a bugfix soon. I can let you know if I continue experiencing the issue once that happens. Thanks!

**Comment by chrmarti** [comment]
Merging with #2193.
